### PR TITLE
objfw: init at 1.1.7

### PR DESCRIPTION
**Pull request by steeleduncan**
ObjFW is one of the Objective-C runtimes supported by clang

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ob/objfw/package.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "objfw";
+  version = "1.1.7";
+
+  src = fetchFromGitHub {
+    owner = "ObjFW";
+    repo = "ObjFW";
+    rev = "f33bd5b5dec1c30ba0a1f0d8ad8fbfa15a6ef3ff";
```
Comment (NotAShelf) [inline pkgs/by-name/ob/objfw/package.nix:18]: Could use a tag here.

File: pkgs/by-name/ob/objfw/package.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ./configure --prefix=$out --without-tls
+    make
+  '';
+
+  doCheck = true;
+  checkPhase = "make check";
+
+  installPhase = "make install";
+
+  meta = with lib; {
```
Comment (NotAShelf) [inline pkgs/by-name/ob/objfw/package.nix:40]: No nested `with lib`.

**Comment by steeleduncan** [comment]
@NotAShelf Thanks for the review. I've pushed 2 commits to fix this. Is this ok? ie will they get squashed with the merge? or should I start a new PR with it all in a single commit?

**Comment by NotAShelf** [comment]
You can just squash all commits into one - no need for a new PR.

**Comment by steeleduncan** [comment]
@NotAShelf Thanks! I've split those changes into two commits and updated now

File: pkgs/by-name/ob/objfw/package.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "objfw";
+  version = "1.1.7";
+
+  src = fetchFromGitHub {
+    owner = "ObjFW";
+    repo = "ObjFW";
+    rev = "1.1.7-release";
```
Comment (NotAShelf) [inline pkgs/by-name/ob/objfw/package.nix:18]:
```suggestion
    rev = "refs/tags/1.1.7-release";
```

File: pkgs/by-name/ob/objfw/package.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  buildPhase = ''
+    ./autogen.sh
+    ./configure --prefix=$out --without-tls
+    make
+  '';
+
+  doCheck = true;
+  checkPhase = "make check";
+
+  installPhase = "make install";
```
Comment (NotAShelf) [inline pkgs/by-name/ob/objfw/package.nix:38]: I'm not really sure about this section. Though it is not my area of expertise, so I don't want to give you a blocking review either.

For now:
1. Move `./autogen.sh` and `./configure` to `configurePhase`. Add `runHook preConfigure` and `runHook postConfigure` at the beginning and at the end of the phase respectively.
2. Remove buildPhase, `make` should be invoked by mkDerivation automatically.
3. You can try omitting installPhase as well, but I am not very sure.

**Review by NotAShelf — COMMENTED** [review]
Thank you for reminding me this, it had gotten lost in my notifications.

I'm leaving two review comments, but neither blocking nor approving as I'd rather have someone who works with make take a look at this.

File: pkgs/by-name/ob/objfw/package.nix (hunk 3 above)
Comment (steeleduncan) [reply pkgs/by-name/ob/objfw/package.nix]: Thanks, done!

File: pkgs/by-name/ob/objfw/package.nix (hunk 4 above)
Comment (steeleduncan) [reply pkgs/by-name/ob/objfw/package.nix]: Thanks, I've done those. It works fine without the `installPhase`, so I've omitted that

**Comment by steeleduncan** [comment]
@NotAShelf Thanks for the review. Is there someone specific I should ping once the tests have run again? or is it ok to just leave in the queue?

**Comment by NotAShelf** [comment]
I would suggest posting this in the nixpkgs review requests Matrix room. Github notifications usually get lost.

File: pkgs/by-name/ob/objfw/package.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  nativeBuildInputs = [
+    clang
+    automake
+    autogen
+    autoconf
+  ];
+
+  configurePhase = ''
+    runHook preConfigure
+    ./autogen.sh
```
Comment (Artturin) [inline pkgs/by-name/ob/objfw/package.nix:31]: Move the autogen to preConfigure

File: pkgs/by-name/ob/objfw/package.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    ./configure --prefix=$out --without-tls
```
Comment (Artturin) [inline pkgs/by-name/ob/objfw/package.nix:32]: Remove this and add `--without-tls` to `configureFlags`
Prefix among other flags are added by mkderivation so the overriding whole phases should be avoided

File: pkgs/by-name/ob/objfw/package.nix [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    runHook postConfigure
+  '';
+
+  doCheck = true;
+  checkPhase = "make check";
```
Comment (Artturin) [inline pkgs/by-name/ob/objfw/package.nix:37]: That's the default (or `test` depending on the conditional https://github.com/NixOS/nixpkgs/blob/d31617bedffa3e5fe067feba1c68b1a7f644cb4f/pkgs/stdenv/generic/setup.sh#L1485)

Comment (steeleduncan) [reply pkgs/by-name/ob/objfw/package.nix]: Thanks for the review. I've integrated these changes now, and am awaiting OfBorg!

**Review by NotAShelf — APPROVED** [review]

File: pkgs/by-name/ob/objfw/package.nix [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ObjFW";
+    repo = "ObjFW";
+    rev = "refs/tags/1.1.7-release";
+    hash = "sha256-0ylG/2ZSO3b8zdh6W9QJH5OJW9V344CCik1DduV5mhI=";
+  };
+
+  nativeBuildInputs = [
+    clang
```
Comment (Artturin) [inline pkgs/by-name/ob/objfw/package.nix:23]: Missed this, to use clang you should do `clangStdenv.mkDerivation`

Comment (steeleduncan) [reply pkgs/by-name/ob/objfw/package.nix:23]: Ah, thanks, I didn't realise that. Should I open a new PR for this? or should I do that in a future PR when I version bump?

Comment (Artturin) [reply pkgs/by-name/ob/objfw/package.nix:23]: Open a new pr

**Pull request merged**